configure.ac

```diff
@@ -29,6 +29,11 @@ fi
     
 AC_PATH_TOOL(PKG_CONFIG, pkg-config)
 
+# no .pc for GMP
+AC_SEARCH_LIBS([__gmpz_init], [gmp])
+# pkg-config file Nettle doesn't provide hogweed in libs
+AC_SEARCH_LIBS([nettle_rsa_public_key_clear], [hogweed])
```

Comment:
These seem like hacks for static linking rather than something needed for a "normal" link?

Reply:
I'd say leave it out. People are generally expected to provide their own link information when they are doing static linking.

Reply:
These checks aren't  for static linking. They are primarily used for dynamic linking.
Check output of the following command:
```bash
 pkg-config --libs nettle
````

My distro shows just one dep: 
````bash
-lnettle
````

Which is obviously not enough for proper linking for program using Nettle (lacks -lhogweed).
Same applies to GMP case.

Or I'm missing something here?

Reply:
What errors are you getting?

Reply:
Try removing Nettle related string and execute:
````
 ./bootstrap && ./configure && make
````

I'm getting:
````
/usr/bin/ld: ssl.o: undefined reference to symbol 'nettle_rsa_public_key_clear@@HOGWEED_4'
/usr/lib/libhogweed.so.4: error adding symbols: DSO missing from command line
````

Reply:
Alright, the problem is that we are using hogweed methods and not just nettle methods. They are somewhat independent from what I can see in the documentation.

hogweed has a pkg-config file, so please update the configure script to also include that.

Reply:
Ok. You're right. I'll fix this one.
What about GMP check? Can I leave it as it is now?

Reply:
Not really. Do we call any gmp functions directly? If so we can have the check, but we should check for a more sensible function rather than something internal like `__gmpz_init`.

Reply:
Check rdssl_rsa_encrypt() in ssl.c for mpz_* functions.
Even it starts with __ It's not internal function btw:
````
$ readelf -a /usr/lib/libgmp.so|grep mpz_init$
    92: 000000000001f280    37 FUNC    GLOBAL DEFAULT   12 __gmpz_init
`````
Bad name, sure.